blocks/location-overview/location-overview.js

```diff
@@ -0,0 +1,94 @@
+import {
+  domEl,
+  div,
+  a,
+  span,
+  h2,
+  p,
+} from '../../scripts/dom-helpers.js';
+
+import {
+  decorateIcons,
+} from '../../scripts/aem.js';
+
+function getLearnMoreIcon() {
+  const learnMoreIcon = domEl('calcite-icon', {
+    icon: 'arrowRight',
+    scale: 's',
+  });
+
+  const iconContainer = div({ class: 'learn-more-icon-container' }, 'Learn more', learnMoreIcon);
+  return iconContainer;
+}
+
+function getPanelIcon(iconPath = '') {
+  const panelIcon = span({
+    class: iconPath || 'location-overview-panel-icon icon icon-2-5d-mesh-48',
+  });
+
+  return panelIcon;
+}
+
+function getContainerContents(hText, pText, panelIconPath) {
+  const panelIcon = getPanelIcon(panelIconPath);
+  const lernMoreIcon = getLearnMoreIcon();
+
+  const overviewCardHeader = h2({
+    class: 'location-overview-card-header',
+  }, hText);
+
+  const overviewCardText = p({
+    class: 'location-overview-card-text',
+  }, pText);
+
+  const contentsWrapper = div({
+    class: 'location-overview-content-wrapper calcite-animate calcite-animate__in-up',
+  }, panelIcon, overviewCardHeader, overviewCardText, lernMoreIcon);
+
+  return contentsWrapper;
+}
+function getLinkWrapper(hText, pText, cardLinkTo) {
+  const linkWrapper = a({
+    class: 'location-overview-link-wrapper',
+    href: cardLinkTo,
+  }, getContainerContents(hText, pText));
+
+  return linkWrapper;
+}
+
+function getLocationOverviewDomElements(cardCount, headingTexts, genericTexts, cardLinks) {
+  // overviewWrapper wraps around all our containers
+  const overviewWrapper = div({ class: 'location-overview-container-wrapper' });
+
+  for (let i = 0; i < cardCount; i += 1) {
+    const card = div({ class: 'location-overview-card-container calcite-mode-dark' }, getLinkWrapper(headingTexts[i], genericTexts[i], cardLinks[i]));
+    overviewWrapper.appendChild(card);
+  }
+
+  return overviewWrapper;
+}
+export default function decorate() {
+  const locationOverviewCards = document.querySelector('.location-overview-wrapper > div');
+
+  const cardHeadingTexts = [];
+  const cardGenericText = [];
+  const overviewCardLinks = [];
```

Comment:
These arrays make the code a bit confusing, why not call getLinkWrapper directly in the first loop for each iteration?

Reply:
On a second note, I think we should rework the content too, I've update the document with a proposal for the block (just added a second block under it): https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B1C8E9C2E-3941-4A04-8C15-0162D8DD8E23%7D&file=location-overview.docx&action=default&mobileredirect=true

Changes:
- h2 created at the content level
- Changed links so that instead of having 3 different links to the same place only "learn more" is a link

This should make the authoring experience a bit closer to "WYSIWYG", do you foresee any issues with it?

![image](https://github.com/user-attachments/assets/48312f26-60ea-4282-9b67-e0f996e93ede)

Reply:
> Changed links so that instead of having 3 different links to the same place only "learn more" is a link
Yes, its better this way, I guess the default import there was like that because everything is wrapped in an anchor tag.

Having a single link makes more sense.

> h2 created at the content level

Let me see how does this look by default

Reply:
> Having a single link makes more sense.

Changed content into single link structure